lib/signs_ui/alerts/state.ex

```diff
@@ -4,87 +4,147 @@ defmodule SignsUi.Alerts.State do
   def active_alert_ids(pid \\ __MODULE__) do
-    GenStage.call(pid, :active_alert_ids)
+    GenServer.call(pid, :active_alert_ids)
   end
 
-  @spec all(GenStage.stage()) :: Display.t()
+  @spec all(SignsUi.Alerts.State) :: Display.t()
   def all(pid \\ __MODULE__) do
-    GenStage.call(pid, :all)
+    GenServer.call(pid, :all)
   end
 
-  @impl GenStage
-  def init(opts) do
-    {:consumer, %{}, opts}
+  @impl GenServer
+  def init(_) do
+    Process.send(self(), :update, [])
+    {:ok, %{}}
   end
 
-  @impl GenStage
+  @impl GenServer
   def handle_call(:active_alert_ids, _from, state) do
     alert_ids = state |> Map.keys() |> MapSet.new()
 
-    {:reply, alert_ids, [], state}
+    {:reply, alert_ids, state}
   end
 
-  @impl GenStage
+  @impl GenServer
   def handle_call(:all, _from, state) do
-    {:reply, Display.format_state(state), [], state}
+    {:reply, Display.format_state(state), state}
   end
 
-  @impl GenStage
-  @spec handle_events([Event.t()], GenStage.from(), t()) :: {:noreply, [], t()}
-  def handle_events(events, _from, state) do
-    # Works in two primary phases. First, we generate fresh t using the
-    # provided events:
-    new_state = update_state(events, state)
-    Logger.info(["alert_state_updated ", inspect(new_state)])
-
-    # Next, we convert our internal model to the specified format:
-    display_state = Display.format_state(new_state)
-    SignsUiWeb.Endpoint.broadcast!("alerts:all", "new_alert_state", display_state)
-    {:noreply, [], new_state}
+  @impl GenServer
+  def handle_info(:update, state) do
+    state =
+      case fetch_alerts() do
+        {:ok, response} ->
+          new_state = Enum.into(parse(response), %{}, &{&1.id, &1})
+
+          SignsUiWeb.Endpoint.broadcast!(
+            "alerts:all",
+            "new_alert_state",
+            Display.format_state(new_state)
+          )
+
+          Logger.info(["alert_state_updated ", inspect(new_state)])
+          new_state
+
+        {:error, error} ->
+          Logger.error([
+            "alerts_fetch_failed, reason=",
+            inspect(error)
+          ])
+
+          state
+      end
+
+    schedule_update(self(), 5_000)
+    {:noreply, state}
   end
 
-  @spec update_state(Event.t() | [Event.t()], t()) :: t()
-  defp update_state(events, state) when is_list(events) do
-    # This reduce combines the effects of a  set of operations into a single new
-    # state.
-    Enum.reduce(events, state, &update_state(&1, &2))
+  defp fetch_alerts do
+    http_client = Application.get_env(:signs_ui, :http_client)
+    url = "#{Application.get_env(:signs_ui, :api_v3_url)}/alerts"
+    headers = [{"x-api-key", Application.get_env(:signs_ui, :api_v3_key)}]
+
+    case http_client.get(
+           url,
+           headers,
+           params: %{
+             "filter[datetime]" => "NOW",
+             "filter[route_type]" => "0,1"
+           }
+         ) do
+      {:ok, %HTTPoison.Response{status_code: 200, body: body}} ->
```

Comment:
Due to load balancing/caching, the API can sometimes return slightly older data on subsequent calls, which means that unless we track `Last-Modified`, we could see this state briefly "out of order" around transitions. That's probably fine for this case, since this data doesn't change very often, and it only populates the form UI, but I wanted to call it out.